### ISSUE TITLE
UnnecessaryParentheses where IDE prompts for clarifying parentheses.

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#4495]
The following code triggers a warning in the IDE, saying `Expression should use clarifying parentheses`:

```
val (adjustedFromItem, fromFragment) =
      validPages
        .firstOrNull { (from, frag) -> from != page && frag.isVisible }
        ?: null to null
```

It wants to correct it to:

```
val (adjustedFromItem, fromFragment) =
      validPages
        .firstOrNull { (from, frag) -> from != page && frag.isVisible }
        ?: (null to null)
```

But detekt flags it as `UnnecessaryParentheses`. I see the value of adding parentheses here, and a few other places that the IDE is suggesting it, but I don't want to have to suppress the detekt check in all of those places.

Can detekt use similar logic to exclude these cases?

**Comment by BraisGabin** [comment]
We should look at the rule that is warning this in IJ and see if we can integrate it in this rule.

**Comment by cortinico** [comment]
> The following code triggers a warning in the IDE

Can we confirm that is a IDE warning and not a compiler warning? A compiler warning would be way easier to integrate if so.

**Comment by eygraber** [comment]
https://youtrack.jetbrains.com/issue/KTIJ-8888

**Comment by dimsuz** [comment]
It seems like this has been fixed by #4881 ?

At least I found the corresponding test case [here](https://github.com/detekt/detekt/blob/d5165629636eb5d37fd10b10761c9363adb808e8/detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParenthesesSpec.kt#L296).

**Comment by BraisGabin** [comment]
Yes, these one was fixed. Closing.

**Comment by eygraber** [comment]
@BraisGabin I found two occurrences where the IDE tells me to add clarifying parentheses, but after I do, detekt says it is a `UnnecessaryParentheses` violation. Here they are after adding the clarifying parentheses:

```
string("id") to (floatOrNull("progress") ?: (index.toFloat() / (segments.size - 1)))
```

and

```
progress = segmentJson.floatOrNull("progress") ?: (index.toFloat() / (segmentArray.size - 1)),
```

**Comment by BraisGabin** [comment]
I was thinking that maybe #5684 would fix it. But after checking a bit more it doesn't seem so.

**Comment by atulgpt** [comment]
Hi @BraisGabin / @eygraber I have added the fix at https://github.com/detekt/detekt/pull/5983
